redash/__init__.py

```diff
@@ -68,71 +60,13 @@ def create_redis_connection():
 
 
 setup_logging()
-redis_connection = create_redis_connection()
-
-mail = Mail()
-migrate = Migrate()
-mail.init_mail(settings.all_settings())
-statsd_client = StatsClient(host=settings.STATSD_HOST, port=settings.STATSD_PORT, prefix=settings.STATSD_PREFIX)
-limiter = Limiter(key_func=get_ipaddr, storage_uri=settings.LIMITER_STORAGE)
 
-import_query_runners(settings.QUERY_RUNNERS)
-import_destinations(settings.DESTINATIONS)
+redis_connection = LocalProxy(create_redis_connection)
 
-from redash.version_check import reset_new_version_status
-reset_new_version_status()
+mail = LocalProxy(Mail)
 
+migrate = LocalProxy(Migrate)
```

Comment:
Why Migrate and Mail need a proxy?

Reply:
This is mostly defensive code, e.g. flask-migrate imports alembic. I'll revert.